src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/ArrayHelpers.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime;
+
+using Debug = System.Diagnostics.Debug;
+
+namespace Internal.Runtime.CompilerHelpers
+{
+    /// <summary>
+    /// Array helpers for generated code.
+    /// </summary>
+    internal static class ArrayHelpers
+    {
+        /// <summary>
+        /// Helper for array allocations via `newobj` IL instruction. Dimensions are passed in as block of integers.
+        /// </summary>
+        private static unsafe Array NewObjArray(IntPtr pEEType, int nDimensions, int* pDimensions)
+        {
+            EETypePtr eeType = new EETypePtr(pEEType);
+            Debug.Assert(eeType.IsArray);
+            Debug.Assert(nDimensions > 0);
+
+            if (eeType.IsSzArray)
```

Comment:
Jagged SzArrays have synthetic constructors as well - they are synthetized (here)[ https://github.com/dotnet/corert/blob/master/src/Common/src/TypeSystem/Common/ArrayType.cs#L80].

E.g. Run the following on full framework or CoreCLR:

`foreach (var c in typeof(int[][][]).GetConstructors()) Console.WriteLine(c);`

Result:
```
Void .ctor(Int32)
Void .ctor(Int32, Int32)
Void .ctor(Int32, Int32, Int32)
```